test/unittest/tests/block/test_block_other.cpp

```diff
@@ -0,0 +1,142 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * This file unit tests the following block manager functions:
+ *    - __wt_page_header_byteswap
+ *    - __wt_block_header_byteswap_copy
+ *    _ __wt_block_eligible_for_sweep
+ */
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+
+uint16_t
+swap_bytes16(uint16_t i)
+{
+    // byte 2 | byte1.
+    uint16_t byte1 = i & 0x000000ff;
+    uint16_t byte2 = (i & 0x0000ff00) >> 8;
+
+    // byte 1 | byte2.
+    return (byte1 << 8 | byte2);
+}
+
+uint32_t
+swap_bytes32(uint32_t i)
+{
+    // byte4 | byte3 | byte2 | byte1.
+    uint32_t byte1 = i & 0x000000ff;
+    uint32_t byte2 = (i & 0x0000ff00) >> 8;
+    uint32_t byte3 = (i & 0x00ff0000) >> 16;
+    uint32_t byte4 = (i & 0xff000000) >> 24;
+
+    // byte 1 | byte2 | byte3 | byte4.
+    return (byte1 << 24 | byte2 << 16 | byte3 << 8 | byte4);
+}
+
+uint64_t
+swap_bytes64(uint64_t i)
+{
+    // byte8 | byte7 | byte6 | byte5 | byte4 | byte3 | byte2 | byte1.
+    uint64_t byte1 = i & 0x00000000000000ffUL;
+    uint64_t byte2 = i & 0x000000000000ff00UL;
+    uint64_t byte3 = i & 0x0000000000ff0000UL;
+    uint64_t byte4 = i & 0x00000000ff000000UL;
+    uint64_t byte5 = i & 0x000000ff00000000UL;
+    uint64_t byte6 = i & 0x0000ff0000000000UL;
+    uint64_t byte7 = i & 0x00ff000000000000UL;
+    uint64_t byte8 = i & 0xff00000000000000UL;
+
+    // byte1 | byte2 | byte3 | byte4 | byte5 | byte6 | byte7 | byte8.
+    return (byte1 << 56 | byte2 << 40 | byte3 << 24 | byte4 << 8 | byte5 >> 16 | byte6 >> 24 |
+      byte7 >> 40 | byte8 >> 56);
+}
+
+static void
+test_block_header_byteswap_copy(WT_BLOCK_HEADER *from)
+{
+    WT_BLOCK_HEADER expected, to;
+
+#ifdef WORDS_BIGENDIAN
+    // Swap and copy the relevant block header bytes.
+    expected.checksum = swap_bytes32(from->checksum);
```

Comment:
I was thinking, could we do something similar to the address_pack and address_unpack function where we have hardcoded cases + using swap_bytes32 as well? 

I would also directly just call __wt_bswap32, since it's doing the same thing anyway.

Reply:
Yeah, I'll probably restructure these tests so that it adopts a similar structure. Thanks for the suggestion.